pytorch_lightning/trainer/callback_hook.py

```diff
@@ -52,6 +53,8 @@ def teardown(self, stage: Optional[str] = None) -> None:
         for callback in self.callbacks:
             callback.teardown(self, self.lightning_module, stage=stage)
 
+        self.data_connector.detach_data(self.lightning_module)
+
```

Comment:
This should be in `Trainer._call_teardown_hook`

Reply:
moved again :D